### PR TITLE
Topic/chimney swap example

### PR DESCRIPTION
**Pull request by robinsdm**

### CODE REVIEW

**Comment by aj-michael** [comment]
I will take a look at this tonight

File: examples/ChimneySwap/.gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+*.iml
+.gradle
+/local.properties
+/.idea/workspace.xml
+/.idea/libraries
+.DS_Store
+/build
+/captures
+/app/google-services.json
+/app/src/main/res/values/api_keys.xml
```
Comment (aj-michael) [inline examples/ChimneySwap/.gitignore:10]: I'm not sure how nested `.gitignore`s work, but you did not remove this file.

File: examples/ChimneySwap/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+
+
+
+
+
+
+
+apply plugin: 'com.google.gms.google-services'
```
Comment (aj-michael) [inline examples/ChimneySwap/app/build.gradle]: Put this with the other plugin and remove excess whitespace

File: examples/ChimneySwap/app/proguard-rules.pro [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Add project specific ProGuard rules here.
```
Comment (aj-michael) [inline examples/ChimneySwap/app/proguard-rules.pro]: Do we need this file? There is nothing in it.

File: examples/ChimneySwap/app/src/androidTest/java/xyz/thepathfinder/www/chimneyswap/ApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package xyz.thepathfinder.www.chimneyswap;
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/androidTest/java/xyz/thepathfinder/www/chimneyswap/ApplicationTest.java]: Is this file needed?

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+public class Chimney {
+
+    private int id;
+    private String name;
+    private String imageUrl;
+    private Bitmap image;
+    private LatLng position;
+
+    public Chimney() {
+        this.id = 0;
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java]: This line and the 4 lines following it are no-ops

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import com.android.volley.toolbox.JsonObjectRequest;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.UnsupportedEncodingException;
+
+public class EasyJsonObjectRequest extends JsonObjectRequest {
+
+    private static final String TAG = "EasyJsonObjectRequest";
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java]: This is a matter of preference, but if you used `EasyJsonObjectRequest.class.getName()` instead of the hardcoded String, refactoring tools (IntelliJ) would catch it if you ever renamed this class.

File: examples/ChimneySwap/gradle.properties [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# Project-wide Gradle settings.
```
Comment (aj-michael) [inline examples/ChimneySwap/gradle.properties:1]: Do we need this file?

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+public class ChimneyAdapter extends ArrayAdapter<Chimney> {
+
+    private final String TAG = "ChimneyAdapter";
+
+    private List<Chimney> chimneys;
+    private Context context;
+
+    public ChimneyAdapter(Context context, List<Chimney> chimneys) {
+        super(context, -1, chimneys);
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java]: What is `-1`? magic number

**Comment by aj-michael** [comment]
It looks good, just added some nitpick comments. Feel free to ignore however many of the comments as you want and tell me to merge  it.

File: examples/ChimneySwap/app/build.gradle [hunk 9]
```diff
@@ -0,0 +1,48 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 23
+    buildToolsVersion "23.0.2"
+
+    defaultConfig {
+        applicationId "xyz.thepathfinder.www.chimneyswap"
```
Comment (aj-michael) [inline examples/ChimneySwap/app/build.gradle]: What's with the `www`?

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+
+public class LoginActivity extends AppCompatActivity {
+
+    private static String TAG = "LoginActivity";
+
+
+    private static final int RC_SIGN_IN = 6124;
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java]: Comment describing what this is? I can't figure it out.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+        setContentView(R.layout.activity_main);
+
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
+
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Thread.sleep(2000);
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java]: What's this for?

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ParcelableChimney.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
+package xyz.thepathfinder.www.chimneyswap;
+
+import android.graphics.Bitmap;
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import com.google.android.gms.maps.model.LatLng;
+
+public class ParcelableChimney implements Parcelable {
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ParcelableChimney.java]: I was about to write a comment saying that you spelled "Parseable" wrong and then realized this is an android thing? lolandroid

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/PostChimneyActivity.java [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+        setContentView(R.layout.activity_post_chimney);
+
+        ParcelableChimney parcelableChimney = getIntent().getParcelableExtra(TradeChimneyActivity.TRADED_CHIMNEY);
+        this.tradedChimney = parcelableChimney.getChimney();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent resultData) {
+        if(requestCode == READ_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            Uri uri = null;
```
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/PostChimneyActivity.java]: Is there any reason you declared this outside the scope of the if-statement?

File: examples/ChimneySwap/.gitignore (hunk 1 above)
Comment (robinsdm) [inline examples/ChimneySwap/.gitignore:10]: It is file a to put the api keys. I just removed the keys to give people the format for it. The .gitignore keeps it from being uploaded with real values.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java (hunk 5 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java]: They are not no-ops, it is explicit initialization. Even if Java will do the same thing without them, we should keep it. Not everyone knows Java's default initialization rules.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java (hunk 8 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java]: -1 keeps the super method from trying to find a TextView.to populate. Since, it includes a picture Android wouldn't be able to auto populate the View.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java (hunk 10 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java]: It is a random number used by Android to route the token back to the correct place. I have no clue why they require you to manager it.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java (hunk 11 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java]: It makes the pretty opening screen stay up for 2 seconds.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ParcelableChimney.java (hunk 12 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ParcelableChimney.java]: Hey, we spelled calendar wrong for the majority of the CSSE221 final project.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java (hunk 10 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java]: Android allows the result of a request to go to a different Activity. So you use these random numbers to make sure it is the result you want and not something another Activity wanted.

File: examples/ChimneySwap/.gitignore (hunk 1 above)
Comment (aj-michael) [inline examples/ChimneySwap/.gitignore:10]: ok

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java (hunk 5 above)
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java]: they are already initialized to those values so it is a no-op. I agree to disagree, but its your SDK :+1:

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java (hunk 8 above)
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/ChimneyAdapter.java]: Please indicate that in the code somehow.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java (hunk 10 above)
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/LoginActivity.java]: ok, comment that

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java (hunk 11 above)
Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/MainActivity.java]: lol nice

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java (hunk 5 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/Chimney.java]: No the Java's compiler is smart enough to remove redundant initialization. Anyway, it is better to not leave ambiguity in the code.

File: examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java (hunk 6 above)
Comment (robinsdm) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java]: Tags are meant to group logging messages not be class names. No Android documentation uses class names

Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java]: Then why are you using class names?

Comment (aj-michael) [inline examples/ChimneySwap/app/src/main/java/xyz/thepathfinder/www/chimneyswap/EasyJsonObjectRequest.java]: Im just saying that this String will not be caught by a refactor. So if we decided that "EasyJsonObjectRequest" was not a descriptive enough class name and refactored it to "ChimneySwapJsonRequest", an IDE will leave that string as "EasyJsonObjectRequest".
